prompter.lisp

```diff
@@ -434,41 +483,37 @@ See `resolve-marks' for a reference on how `marks' are handled."
         (not (actions-on-current-suggestion-enabled-p source))))
 
 (export-always 'next-ready-p)
-(defun next-ready-p (prompter)
+(defun next-ready-p (prompter &key (wait-p t))
   "Block and return next PROMPTER ready source.
 It's the next source that's done updating.
 If all sources are done, return T.
 This is unblocked when the PROMPTER is `destroy'ed."
   (when prompter
-    ;; We let-bind `sync-queue' here so that it remains the same object throughout
-    ;; this function, since the slot is subject to be changed concurrently when
-    ;; the input is edited.
-    (alex:if-let ((sync-queue (sync-queue prompter)))
-      (if (= (length (ready-sources sync-queue))
-             (length (sources prompter)))
-          t
-          (calispel:fair-alt
-            ((calispel:? (ready-channel sync-queue) next-source)
-             (cond
-               ((null next-source)
-                nil)
-               (t
-                (push next-source (ready-sources sync-queue))
-                ;; Update current suggestion when update is done:
-                (first-suggestion prompter)
-                next-source)))
-            ((calispel:? (sync-interrupt-channel sync-queue))
-             nil)))
-      ;; No sync-queue if no input was ever set.
-      t)))
+    (block nil
+      (lpara:task-handler-bind ((lpara:task-killed-error (lambda (c)
```

Comment:
Hmmmm, why doesn't Lparallel have `handler-case` alternative? 🤔

Reply:
`handler-bind` is more general-purpose, as it does not unwind the stack.
In any case, it's the only place where we use this `block nil`, so there... :)